test/ffmpeg-qsv/transcode/avc.py

```diff
@@ -0,0 +1,190 @@
+###
+### Copyright (C) 2018-2019 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ..util import *
+from .transcoder import TranscoderTest
+
+spec = load_test_spec("avc", "transcode")
+class to_avc(TranscoderTest):
+  @slash.requires(have_ffmpeg_h264_qsv_encode)
+  @slash.requires(have_ffmpeg_h264_qsv_decode)
+  @slash.parametrize(*gen_transcode_1to1_parameters(spec, "avc", "hwhw"))
+  @platform_tags(set(AVC_DECODE_PLATFORMS) & set(AVC_ENCODE_PLATFORMS))
+  def test_hwhw_1to1(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'h264',
+      case         = case,
+      mode         = 'hwhw',
+      ffencoder    = 'h264_qsv',
+      ffdecoder    = 'h264_qsv',
+    )
+    self.transcode_1to1()
+ 
+  @slash.requires(have_ffmpeg_x264_encode)
+  @slash.requires(have_ffmpeg_h264_qsv_decode)
+  @slash.parametrize(*gen_transcode_1to1_parameters(spec, "avc", "hwsw"))
+  @platform_tags(AVC_DECODE_PLATFORMS)
+  def test_hwsw_1to1(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'h264',
+      case         = case,
+      mode         = 'hwsw',
+      ffencoder    = 'libx264',
+      ffdecoder    = 'h264_qsv',
+    )
+    self.transcode_1to1()
+ 
+  @slash.requires(have_ffmpeg_h264_qsv_encode)
+  @slash.parametrize(*gen_transcode_1to1_parameters(spec, "avc", "swhw"))
+  @platform_tags(AVC_ENCODE_PLATFORMS)
+  def test_swhw_1to1(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'h264',
+      case         = case,
+      mode         = 'swhw',
+      ffencoder    = 'h264_qsv',
+      ffdecoder    = 'h264',
```

Comment:
Curious, does ffmpeg list this SW decoder with "-decoders"?  If so, a slash.requires would be good.  Same for other SW decoder cases.

Reply:
yes, "ffmpeg -decoders" giving SW decoders also and added slash.requires for them